src/common/buffer.cc

```diff
@@ -1068,7 +1068,9 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
   template<bool is_const>
   void buffer::list::iterator_impl<is_const>::advance(unsigned o)
   {
-    p_off += o;
+    if (__builtin_add_overflow(p_off, o, &p_off)) {
```

Comment:
if we throws here, we won't be able to add `noexcept` specifier to these methods. 

quote from Effective Modern C++
> In a noexcept function, optimizers need not keep the runtime stack in an unwindable state if an exception would propagate out of the function, nor must they ensure that objects in a noexcept function are destroyed in the inverse order of construction should an exception leave the function. Functions with “throw()” exception specifications lack such optimization flexibility,